src/crypto/OutgoingRoomKeyRequestManager.js

```diff
@@ -107,6 +110,79 @@ export default class OutgoingRoomKeyRequestManager {
         });
     }
 
+    /**
+     * Cancel room key requests, if any match the given details
+     *
+     * @param {module:crypto~RoomKeyRequestBody} requestBody
+     *
+     * @returns {Promise} resolves when the request has been updated in our
+     *    pending list.
+     */
+    cancelRoomKeyRequest(requestBody) {
+        return this._cryptoStore.getOutgoingRoomKeyRequest(
+            requestBody,
+        ).then((req) => {
+            if (!req) {
+                // no request was made for this key
+                return;
+            }
+
+            if (req.state === ROOM_KEY_REQUEST_STATES.UNSENT) {
+                // just delete it
+                console.log(
+                    'deleting unnecessary room key request for ' +
+                    stringifyRequestBody(requestBody),
+                );
+                return this._cryptoStore.deleteOutgoingRoomKeyRequest(
+                    req.requestId, ROOM_KEY_REQUEST_STATES.UNSENT,
+                );
+            }
+
+            if (req.state !== ROOM_KEY_REQUEST_STATES.SENT) {
+                // the only other option is cancellation_pending, in which case
+                // there is nothing else to do here
+            }
```

Comment:
>  If I explicitly handle each case, it's still going to break in weird ways if we end up adding more states.

The point is to try to identify new states and fail as quickly as possible rather than executing whichever state was last to be handled. I was hoping for something like:
```js
switch(foo) {
  case 1:
    break;
  case 2:
    break;
  case 3:
    break;
  default:
    throw new Error("unhandled state: " + foo);
}
```

which makes is abundantly clear in logs that we forgot to implement stuff, rather than absolute silence and it sending a cancellation/sending the request/deleting the request/other things you really don't want.